### ISSUE TITLE
support disabling statsd metric key sanitizing

### ISSUE THREAD

**Issue by rdwr-tomers** [vectordotdev/vector#20595]
### A note for the community

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Use Cases

Looking into replacing fluentbit with vector. 

Fluentbit statsd input leaves metric names as is. 
Vector replaces "/" with "-", whitespaces with "_" and deletes all non alphanumeric characters.

parsing the values after these replacements are made is impossible.

### Attempted Solutions

Could not find workaround for this.

### Proposal

add configuration option for 'statsd' source to not sanitize metric name. 
let user use transforms to modify it later, if needed

### Version

vector 0.38.0

**Comment by jszwedko** [comment]
Thanks for filing this @to-m . I agree. StatsD [doesn't seem to](https://github.com/b/statsd_spec?tab=readme-ov-file) have a strong opinion on valid metric names. I might even argue the `statsd` source shouldn't be sanitizing at all and that users should do that in a follow-up `remap` transform if desired.

**Comment by to-m** [comment]
Hi @jszwedko - I whole heartedly agree, but since existing users may have come to rely on the existing behavior I didn't think to suggest to replace it. 
Also - a quick question - do you have any information at all with regards to cost of doing the sanitization in the statsd source vs. in remap? If using remap is more expensive - it would be better to leave it as an option in the statsd source behavior.

**Comment by jszwedko** [comment]
Agreed, given this would be a breaking change, a configuration field to opt out makes sense to me.

Using `remap` is likely to be more expensive compared with native Rust code, but I don't think it'd be prohibitively so. `remap` is designed to be performant.
